api.c: enable controller in the root_cgroup.subtree_file

kamalesh-babulal opened the pull request:
`systemd` by default only enables `cpu, cpuset, io, memory`, and `pids`
controller the `root_cgroup.subtree_control`. Trying to enable other
controllers (`hugetlb` and `misc`) in a nested cgroups create scenario,
will fail because they are not, enabling the controller in the
`root_cgroup.subtree_control` file.
    
```
$ sudo cgcreate -ghugetlb:foo/bar
Error: Failed to delete foo/bar: No such file or directory
cgcreate: can't create cgroup foo/bar: No such file or directory
```
    
Fix this by enabling the controller in the `root_cgroup.subtree_control`
file unconditionally for all controllers, if not already enabled, while
calling `cgroupv2_subtree_control_recursive()` to enable the controller
in the given cgroup and its descendants. Checking and enabling every
controller unconditionally should work in the future, in case systemd
disables some other controllers too.

This patchset has two other patches, that update the `gunit/012` test
to adopt the new checks and add new `ftests` test cases that will recreate
the nested cgroups with `hugetlb` and `misc` controllers.

Fixes: #405

drakenclimber commented on tests/ftests/090-cgcreate-non-default-controllers-v2.py:
```diff
@@ -0,0 +1,85 @@
+CONTROLLERS = ['hugetlb', 'misc']
+
+CGNAME = '090cgcreate'
+CGNAME1 = '{}/childcg'.format(CGNAME)
```
nitpick - you could use `os.path.join()` here.

kamalesh-babulal replied:
@drakenclimber Thank you so much for the review.  Sure, will use `os.path.join()` here.

kamalesh-babulal commented on tests/ftests/090-cgcreate-non-default-controllers-v2.py:
```diff
@@ -0,0 +1,85 @@
+    result = consts.TEST_PASSED
+    cause = None
+
+    # checking for childcg controllers, will check in 090cgcreate.subtree_control file
```
Ah !! It sounds incorrect, how about this:
```
checking if the controller is enabled in granchildcg, in turn will check its parent cgroup childcg's subtree_control file, if it's enabled in parent cgroup, it's also enabled in the grandparent too.
```

kamalesh-babulal commented:
**v2 changes**:
- Addressed the review comments from @drakenclimber 
1. Used `os.path.join()` in the ftests testcase
2. Improved  the comments in the ftests testcase
- Renumbered the test case from `090` to `091`, as they are reserved for `cgset` and `cgxset` recursive test.

drakenclimber commented:
Pushed to main.  Thanks, @kamalesh-babulal
